### PR TITLE
Add a State Migration Troubleshooting Guide

### PR DESCRIPTION
**Pull request by Constantine19**
I've included a quick Troubleshooting Guide that can be updated as needed to address any issues we encounter during the migration process.

### CODE REVIEW

File: TFC/migrate-state/troubleshooting-guide.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        - Re-export all variables if needed
+     4. Re-run the migration script
+
+3. **Error Acquiring State Lock:**
+   - If you encounter an "Error acquiring the state lock":
+     - Copy the lock ID from the message
+     - Go to the env0 platform 
+        - Project environment -> kebab menu -> Run a task:
+            ```
+            terraform force-unlock
```
Comment (sagilaufer1992) [inline TFC/migrate-state/troubleshooting-guide.md:41]: shouldn't there be a force lock id here too?
```suggestion
            terraform force-unlock -force <LOCK_ID>
```

File: TFC/migrate-state/troubleshooting-guide.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        - Remove all generated workspace files
+        - Re-run the migration script
+
+2. **Terraform Version Mismatch:**
+   - If you receive a Terraform version mismatch message:
+     1. Install tfenv using: 
+        ```
+        brew install tfenv
+        ```
+     2. Use version `1.5.3`
```
Comment (sagilaufer1992) [inline TFC/migrate-state/troubleshooting-guide.md:26]: the version shouldn't be specific, but match the one in their workspace.
users shouldn't face those, as long as they run a terraform version<1.6.0

File: TFC/migrate-state/troubleshooting-guide.md (hunk 1 above)
Comment (Constantine19) [reply TFC/migrate-state/troubleshooting-guide.md]: copy paste gone wrong :) fixed

File: TFC/migrate-state/troubleshooting-guide.md (hunk 2 above)
Comment (Constantine19) [reply TFC/migrate-state/troubleshooting-guide.md]: yes, I added a clarification that it's only as an example

File: TFC/migrate-state/troubleshooting-guide.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+     - Go to the env0 platform 
+        - Project environment -> kebab menu -> Run a task:
+            ```
+            terraform force-unlock -force <LOCK_ID>
+            ```
+     - If unsuccessful, navigate to the workspace directory and run:        
+        ```
+        terraform force-unlock -force <LOCK_ID>
+        ```
+     - If there's a "Failed to load state" add `-ignore-remote-version` parameter.
```
Comment (sagilaufer1992) [inline TFC/migrate-state/troubleshooting-guide.md:47]: that flag doesn't exist on that command

Comment (Constantine19) [reply TFC/migrate-state/troubleshooting-guide.md]: I saw that we used it in the recording and did not notice that it yielded an invalid flag error

**Review by sagilaufer1992 — APPROVED** [review]
Looks right

**Pull request merged**